### ISSUE TITLE
Func compiler incorrectly handles if operator with a return in the body, when the operator is nested in a loop

### ISSUE THREAD

**Issue by medved239** [ton-blockchain/ton#140]
The FunC code 
```
int test() method_id {
  int i = 0;
  int f = true;
  while (f) {
    i = i + 1;
    if (i > 5) {
      return 1;
    }
    f = (i * i != 64);
  }
  return -1;
}
```
gets compiled into the following Fift assembly (with -O0 flag)
```
PROGRAM{
  105222 DECLMETHOD test
  test PROC:<{
    // 
    0 PUSHINT	//  i=0
    TRUE	//  i=0 f
    WHILE:<{
    }>DO<{	//  i
      INC	//  i
      DUP
      OVER	//  i i i
      MUL	//  i _9
      64 NEQINT	//  i f
    }>	//  i
    DROP	// 
    -1 PUSHINT	//  _12=1
  }>
}END>c
```
As one can see, compiler has simply ignored the if statement. Indeed, the code without the if is compiled into the same Fift assembly. As a consequence, when invoked the code leaves a -1 on the stack, not a 1, as one may expect.
Probably it is related to the error "if and else branches should both return or both not return", but it is not displayed by func compiling the code.

**Comment by Walbr** [comment]
try
`int test() method_id {
  int i = 0;
  int f = true;
int result = -1;
  while (f) {
    i = i + 1;
    if (i > 5) {
      result = 1;
      f = false;
    }
    f = (i * i != 64);
  }
  return result;
}`

**Comment by ton-blockchain** [comment]
Thanks, this was indeed a manifestation of a more serious bug in the funC compiler, fixed in commit b978e27b2f2ca6e4403843b9d6bf3f407bf63499. Now the original code would produce the error "if and else branches should both return or both not return", which is supposed to be the correct behavior at the moment.

If you really need conditional `return`s or `break`s inside a loop, use the trick suggested by Walbr above, that transforms such code into purely structural.
